From 29362213c1f7ff0d1f04cb84a5743137c3cbd1fb Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Tue, 15 Oct 2013 15:07:45 +0200
Subject: [PATCH] cellrenderertext: Fix possible overwriting of markup

When creating the layout, attributes were inserted after existing attributes
with a matching start_index; this could result in some attributes from markup
being ignored/overwritten.

For example, if the markup started with a span setting a color, it would be
overwritten by the foreground property for that section.

Inserting attributes before will ensure markup attributes are preserved.

Signed-off-by: Olivier Brunel <jjk@jjacky.com>
---
 gtk/gtkcellrenderertext.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/gtkcellrenderertext.c b/gtk/gtkcellrenderertext.c
index fc17b38..6f39ff3 100644
--- a/gtk/gtkcellrenderertext.c
+++ b/gtk/gtkcellrenderertext.c
@@ -1577,7 +1577,7 @@ add_attr (PangoAttrList  *attr_list,
   attr->start_index = 0;
   attr->end_index = G_MAXINT;
   
-  pango_attr_list_insert (attr_list, attr);
+  pango_attr_list_insert_before (attr_list, attr);
 }
 
 static PangoLayout*
-- 
1.9.2

